pkg/autoscaling/types.go

```diff
@@ -138,3 +140,46 @@ func (t *TiDBInfo) getLabelValue(key string) string {
 func (t *TiDBInfo) getLabels() map[string]string {
 	return t.Labels
 }
+
+// GetTiDB get TiDB info which registered in PD by address
+func GetTiDB(etcdClient *clientv3.Client, address string) (*TiDBInfo, error) {
+	key := fmt.Sprintf("/topology/tidb/%s/info", address)
+	resp, err := etcdutil.EtcdKVGet(etcdClient, key)
+	if err != nil {
+		return nil, err
+	}
+	if resp.Count < 1 {
+		err := fmt.Errorf("resp loaded for tidb[%s] is empty", address)
+		return nil, err
+	}
+	tidb := &TiDBInfo{}
+	err = json.Unmarshal(resp.Kvs[0].Value, tidb)
+	if err != nil {
+		return nil, err
+	}
+	tidb.Address = address
+	return tidb, nil
+}
+
+// GetTiDBs list TiDB register in PD
+func GetTiDBs(etcdClient *clientv3.Client) ([]*TiDBInfo, error) {
+	tidbInfoPattern, err := regexp.Compile("/topology/tidb/.+/info")
```

Comment:
prefer to use a const for "/topology/tidb/.+/info"

Reply:
updated.